src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -126,118 +126,6 @@ namespace {
     return true;
   }
 
-  void check_spatial_eq(SpatialPooler sp1, SpatialPooler sp2)
-  {
-    UInt numColumns = sp1.getNumColumns();
-    UInt numInputs = sp2.getNumInputs();
-
-    ASSERT_TRUE(sp1.getNumColumns() == sp2.getNumColumns());
-    ASSERT_TRUE(sp1.getNumInputs() == sp2.getNumInputs());
-    ASSERT_TRUE(sp1.getPotentialRadius() ==
-              sp2.getPotentialRadius());
-    ASSERT_TRUE(sp1.getPotentialPct() == sp2.getPotentialPct());
-    ASSERT_TRUE(sp1.getGlobalInhibition() ==
-              sp2.getGlobalInhibition());
-    ASSERT_TRUE(sp1.getNumActiveColumnsPerInhArea() ==
-              sp2.getNumActiveColumnsPerInhArea());
-    ASSERT_TRUE(almost_eq(sp1.getLocalAreaDensity(),
```

Comment:
TODO: many uses of `almost_eq()` can be replaced with `EXCEPT_NEAR(a, b, diff)`; and the function can be removed from SPTest class.
https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md

Although, for the comparisons we need a boolean, not an assert; so we'd have to keep the function?

Reply:
We'll just implement the small `almost_eq` method ; see "twice" in the example
https://github.com/numenta/nupic.core/pull/1384#discussion_r160955339